Python/dynload_shlib.c

```diff
@@ -38,12 +42,18 @@ const char *_PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
     ".dll",
 #else  /* !__CYGWIN__ */
-    "." SOABI ".so",
-#ifdef ALT_SOABI
-    "." ALT_SOABI ".so",
-#endif
-    ".abi" PYTHON_ABI_STRING ".so",
-    ".so",
+// TARGET_OS_IPHONE covers any non-macOS Apple platform.
+#  if defined(__APPLE__) && TARGET_OS_IPHONE
+#    define SHLIB_SUFFIX ".dylib"
+#  else
+#    define SHLIB_SUFFIX ".so"
+#  endif
```

Comment:
Out of curiosity, is there really strictly no way any extension module could have a `.so` suffix?  What about in places the app wouldn't be expected to look but might still have access to (if that's even possible)?

(To be clear, I'm not opposed to restricting the suffix here if that satisfies conforming to the iOS app rules.)

Reply:
That's a good question - I don't think I've actually tried. iOS App Store policies definitely restrict the *type* of binary you can use in a framework, and I assumed that would extend to using the default extension as well - but this might not be true. It's definitely worth confirming, though - I agree it would be a lot nicer if we could use the ``.so`` extension consistently.

Reply:
So... I feel like a bit of an idiot. It doesn't matter what file extension is used when the module is compiled - we can rename it when it's moved to the final Frameworks location. The extension referenced in this list is only used for finder purposes... but the FileFinder won't ever find a `.so` or `.dylib` file in the normal PYTHONPATH, because the App Store prohibits that.

For bonus facepalm points: the default extension used by an Apple framework is... no extension at all - so when the binary is moved, we can strip off any extension, and rename it to the fully-qualified Python module name.